File: lib/domain/usecases/repository/delete_repository_usecase.dart
import '../../repositories/i_repository_repository.dart';

/// 删除仓库用例
class DeleteRepositoryUseCase {
  DeleteRepositoryUseCase(this._repository);

  final IRepositoryRepository _repository;

  Future<void> execute(String id) async {
    final existing = await _repository.getById(id);
    if (existing == null) {
      throw ArgumentError('仓库不存在: $id');
    }

    await _repository.delete(id);
  }
}

File: web/src/application/usecases/repository/delete-repository-usecase.ts
import { IRepositoryRepository } from '../../../domain/repositories/i-repository-repository';
import { IBranchRepository } from '../../../domain/repositories/i-branch-repository';
import { ITaskRepository } from '../../../domain/repositories/i-task-repository';

export class DeleteRepositoryUseCase {
  constructor(
    private repositoryRepo: IRepositoryRepository,
    private branchRepo: IBranchRepository,
    private taskRepo: ITaskRepository
  ) {}

  async execute(id: string): Promise<void> {
    const branches = await this.branchRepo.getByRepositoryId(id);
    for (const branch of branches) {
      const tasks = await this.taskRepo.getByBranchId(branch.id);
      for (const task of tasks) {
        await this.taskRepo.delete(task.id);
      }
      await this.branchRepo.delete(branch.id);
    }
    await this.repositoryRepo.delete(id);
  }
}
